src/configSchema.ts

```diff
@@ -0,0 +1,225 @@
+	fuzzySizeThreshold: "fuzzySizeThreshold must be between 0 and 1.",
+	injectUrl:
+		"You need to specify rtorrentRpcUrl, transmissionRpcUrl, qbittorrentUrl, or delugeRpcUrl when using 'inject'",
+	dataBased:
+		"Data-Based Matching requires linkType, dataDirs, and linkDir to be defined.",
+	riskyRecheckWarn:
+		"It is strongly recommended to not skip rechecking for risky matching mode.",
+	windowsPath: `Your path is not formatted properly for Windows. Please use "\\\\" or "/" for directory separators.`,
+};
+
+/**
+ * custom zod error map for logging
+ * @param error ZodIssue messaging object
+ * @param ctx ZodError map
+ * @returns (the custom error for config display)
+ */
+export const zodErrorMap: z.ZodErrorMap = (error, ctx) => {
+	switch (error.code) {
+		case z.ZodIssueCode.invalid_union:
+			return {
+				message: error.unionErrors
+					.reduce((acc, error) => {
+						// @ts-expect-error avoids type conflict string to never
+						error.errors.forEach((x) => acc.push(x.message));
+						return acc;
+					}, [])
+					.join("; "),
+			};
+	}
+
+	return { message: ctx.defaultError };
+};
+
+/**
+ * helper function for validation
+ * @return transformed duration (string -> milliseconds)
+ */
+
+function transformDurationString(durationStr: string, ctx) {
+	const duration = ms(durationStr);
+	if (isNaN(duration)) {
+		// adds the error to the Zod Issues
+		ctx.addIssue({
+			code: "custom",
+			message: ZodErrorMessages.vercel,
+		});
+	}
+	return duration;
+}
+
+/**
+ * adds an issue in Zod's error mapped formatting
+ * @param setting the value of the setting
+ * @param errorMessage the error message to append on a newline
+ * @param ctx ZodError map
+ */
+function zodAddIssue(setting: string, errorMessage: string, ctx): void {
+	ctx.addIssue({
+		code: "custom",
+		message: `\t"${setting}"\n\t${errorMessage}`,
+	});
+}
+
+/**
+ * helper function for directory validation
+ * @return true if valid formatting
+ */
+function checkValidPathFormat(paths: string[] | string, ctx) {
+	if (!paths) {
+		return undefined;
+	}
+
+	if (typeof paths === "object") {
+		paths.forEach((pathStr) => {
+			if (
+				sep === "\\" &&
+				!pathStr.includes("\\") &&
+				!pathStr.includes("/")
+			) {
+				zodAddIssue(pathStr, ZodErrorMessages.windowsPath, ctx);
+			}
+		});
+	} else {
+		if (sep === "\\" && !paths.includes("\\") && !paths.includes("/")) {
+			zodAddIssue(paths, ZodErrorMessages.windowsPath, ctx);
+		}
+	}
+	return paths; // This will return the array with only valid paths
+}
+
+/**
+ * an object of the zod schema
+ * each are named after what they are intended to validate
+ */
+
+export const VALIDATION_SCHEMA = z
+	.object({
+		delay: z
+			.number()
+			.positive({
+				message: ZodErrorMessages.delay,
+			})
+			.default(10),
+		torznab: z.array(z.string().url()),
+		dataDirs: z
+			.array(z.string())
+			.transform((value, ctx) =>
+				value && value.length > 0
+					? checkValidPathFormat(value, ctx)
+					: null
+			)
+			.nullish(),
```

Comment:
I initially did this, but it ended up producing errors for `dataDirs,#` for its position in the array, and my alternative with that was to `split()` and stuff on the actual error'ing output how I had previously done with fuzzysizethreshold. I did not particularly like that, as it convoluted the error messaging.

It's not particularly a big deal, but this way it just uses dataDirs, and there is no ternary's or parsing done currently with the logging of errors from Zod. I know it sort of makes the checkValidPathFormat messy, but that shouldn't really need much attention (if any) going forward since it's pretty much a static thing.

I'll wait to hear what you think.

Reply:
Just pushed handling arrays with a "position" distinction for the error message.

It's kind of messy, but it will work for any array we ever decide to employ, so I think it's good.

Reply:
I agree, I like that it'll work for any array we decide to employ.